features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/EditParameter.jsx

```diff
@@ -72,13 +77,21 @@ export default function EditParameter(props) {
      */
     function parameterActionReducer(currentParameter, paramAction) {
         const { action, value } = paramAction;
-        const nextParameter = { ...currentParameter, schema: { ...currentParameter.schema } };
+        const nextParameter = currentParameter.schema
+            ? { ...currentParameter, schema: { ...currentParameter.schema } } : { ...currentParameter };
         switch (action) {
             case 'description':
             case 'required':
                 nextParameter[action] = value;
                 break;
-
+            case 'type':
+            case 'format':
+                if (nextParameter.schema) {
```

Comment:
Why we check for nextParameter.**schema** in here and if it's not available directly assign value to nextParameter[action] ? is it because of a difference in Swagger/OAS spec versions